crates/webidl/src/lib.rs

```diff
@@ -210,19 +210,38 @@ impl<'a> WebidlParse<&'a webidl::ast::NonPartialInterface> for webidl::ast::Exte
     ) -> Result<()> {
         let mut add_constructor = |arguments: &[webidl::ast::Argument], class: &str| {
             let self_ty = ident_ty(rust_ident(&interface.name));
+
             let kind = backend::ast::ImportFunctionKind::Method {
                 class: class.to_string(),
                 ty: self_ty.clone(),
                 kind: backend::ast::MethodKind::Constructor,
             };
+
+            let structural = false;
+
+            // Constructors aren't annotated with `[Throws]` extended attributes
+            // (how could they be, since they themselves are extended
+            // attributes?) so we must conservatively assume that they can
+            // always throw.
+            //
+            // From https://heycam.github.io/webidl/#Constructor (emphasis
+            // mine):
+            //
+            // > The prose definition of a constructor must either return an IDL
+            // > value of a type corresponding to the interface the
+            // > `[Constructor]` extended attribute appears on, **or throw an
+            // > exception**.
+            let throws = true;
```

Comment:
This seems a little unfortunate :(

I wonder though if we can add configuration options to the generation to handle this? I'd expect that the majority of webidl constructors don't actually throw (or we're not interested in what they throw). We could perhaps explicitly whitelist some constructors as throwing but all others would be infallible?

Reply:
We could certainly add our own `[WbgNoThrow]` extended attribute if we get to a place where we want  to explicitly whitelist some constructors.

Reply:
Perhaps yeah, although I was thinking moreso the build script rather than modifying the webidl files themselves, but either way's fine!

Reply:
It would be nice if WebIDL had a `NoThrowsConstructor` or something like that.

I think for now, we should probably just leave this as is, and maybe come back to some whitelist thing -- maybe make an issue about it?

Reply:
I'll file a follow up.

Reply:
https://github.com/rustwasm/wasm-bindgen/issues/458